lib/models/application.coffee

```diff
@@ -301,13 +338,20 @@ getApplicationModel = (deps, opts) ->
 	# });
 	#
 	# @example
+	# resin.models.application.getApiKey(123).then(function(apiKey) {
+	# 	console.log(apiKey);
+	# });
+	#
+	# @example
 	# resin.models.application.getApiKey('MyApp', function(error, apiKey) {
 	# 	if (error) throw error;
 	# 	console.log(apiKey);
 	# });
 	###
-	exports.getApiKey = (name, callback) ->
-		exports.get(name).then (application) ->
+	exports.getApiKey = (nameOrId, callback) ->
+		# Do a full get, not just getId, because the actual api endpoint doesn't fail if the id
+		# doesn't exist. TODO: Can use getId once https://github.com/resin-io/resin-api/issues/110 is resolved
+		exports.get(nameOrId).then (application) ->
```

Comment:
Would be nice to pass the `$select: 'id'` option to speedup things.
Would also rename the method to `generateApiKey` or `createApiKey`, the name is confusing.